clang/include/clang/Driver/Options.td

```diff
@@ -2887,6 +2887,10 @@ def fxray_selected_function_group :
   HelpText<"When using -fxray-function-groups, select which group of functions to instrument. Valid range is 0 to fxray-function-groups - 1">,
   MarshallingInfoInt<CodeGenOpts<"XRaySelectedFunctionGroup">, "0">;
 
+def fxray_enable_shared : Flag<["-"], "fxray-enable-shared">, Group<f_Group>,  Visibility<[ClangOption, CC1Option]>,
```

Comment:
Use `BoolFOption`. You can check some existing ones and how they determine whether `clang -cc1 -ffoo` or `clang -cc1 -fno-foo` is available.

Reply:
Thanks, I changed the option as requested